Rethinking RFCs (#4817)
sourcegraph/about: handbook/communication/rfcs/index.md

> # Requests for Comments (RFCs)

Reviewer: nit: just curious why you chose to capitalize "Comments" here. The [style guide](https://about.sourcegraph.com/handbook/communication/content_guidelines/style_and_mechanics#capitalization) would indicate that it should remain lowercase.

Author: ```suggestion
# Requests for comments (RFCs)
```

Essentially I was thinking of it as a "name," but don't feel strongly one way or the other. Same with "Tension"—but I think it feels more internally consistent (and with the styleguide) with your suggested changes applies, so doing so!

> We use RFCs to collaborate successfully as an [all-remote](../../company/remote/index.md) team.

Reviewer: I don't know where exactly to fit this in, but I think it might be valuable to have a quickstart section a the top that let's someone, who is already familiar with RFCS, quickly create a new RFC.

```markdown
## Quickstart

* [Folder of public RFCs](#link)
* [Folder of private RFCs](#link)
* Create a new RFC
  * [Default format](https://docs.google.com/document/d/1VV0ddLmMrcU2IWo_s4xm8Q8UtGnreyuhIRPJY0Dh5NI/copy)
  * (list other formats)
```

 (note that you can have a google doc link end with `/copy` to prompt the user to make their own copy)

Author: Added!

Thanks for the tip on `/copy` – I had no idea!

RFCs give us a way to write down ideas and plans so we can communicate, collect thoughtful feedback from others on the team, and [make decisions](../decisions.md).

An “RFC” literally means a “Request for Comments.” Though RFCs at Sourcegraph can take many forms for different purposes, an RFC is ultimately just a document with a few attributes:

- The author’s name(s)
- The date it was written
- A sequential number

> - A status label

Reviewer: I am wondering if we should bias more towards status prose in the doc rather than labels in the title.

Reviewer 2: I think there is a lot of value to be had from having the status in the title. Since we don't use a structured format for our RFC (like Markdown documents in a repo) it's not easy to extract metadata like status in tooling. Take for example the value you get when using `sg rfc list` by having it in the title.
<img width="1147" alt="screen-capture--2021-09-28--0918@2x" src="https://user-images.githubusercontent.com/492942/135041583-8d6e54a3-223d-46e6-a250-f8147716472a.png">

Author: Based on this feedback, let's keep the status in the title until we have some other solution to exposing better metadata.

- Is public to the world by default
- Is a Google Doc

Our goal is for RFCs to be lightweight, low-process, and effortless to create and use. An RFC can be thought of as an asynchronous conversation. Unlike Slack, email, or other channels, using RFCs helps us because:

- There’s no implicit demand for an immediate response.
- Reviewers have time to consider and propose changes.
- More people can collaborate at once without clashing.
- RFCs are easily searchable and referable.
- RFCs are retained indefinitely.

## When to use an RFC

### Use an RFC when…

> - You want to drive a decision using our [formal decision-making process](../decisions.md).

Reviewer 3: So I actually don't think this is a good reason to create an RFC. In the vast majority of cases, that process points to "just make the decision yourself and document it". 

```suggestion
```

Author: I agree in spirit here, and think we can add some stuff to make that really clear. I think there's a lot of benefit in using RFCs for the bigger, "highly cross-functional decisions"—of which we had at least one in the past few months.

```suggestion
- You need to surface and communicate around a highly cross-functional decision [formal decision-making process](../decisions.md).
```

- You want to frame a problem and propose a solution.
- You want thoughtful feedback from team members on our globally-distributed remote team.
- You want to surface an idea, tension, or feedback.
- You want to define a project or design brief to drive project collaboration.

### Don’t use an RFC when…

> - You want to discuss personal or sensitive topics one-on-one with another team member.

Reviewer 3: ```suggestion
- You want to discuss personal or sensitive topics one-on-one with another team member.
- You want to [make a decision](../decisions.md) to change something where you are the decider. In the vast majority of cases, creating an RFC to explain yourself will be overkill. RFCs should only be used if a decision explicitly requires one of the bullets in the section above.
```

These status labels help to provide more context for readers. They’re most helpful for readers to understand the RFC’s context and shape their feedback accordingly.

We have a set of status labels we’ve found helpful in the past:

- **WIP**: The author is still drafting the RFC and it’s not ready for review.
- **Review**: The Review label is used when the RFC is ready for comments and feedback.
- **Approved**: When the RFC is for the purpose of making a decision, the Approved label indicates that the decision has been made.
- **Implemented**: When the RFC is for the purpose of making a decision, the Implemented label indicates that the RFC’s proposal has been implemented.
- **Closed**: When the RFC is for the purpose of collaboration or discussion but not necessarily to make a decision or propose a specific outcome that will eventually become Implemented, the Closed label indicates that the RFC is no longer an active collaborative artifact.
- **Abandoned**: When the RFC is for the purpose of making a decision, and there are no plans to move forward with the RFC’s proposal, the Abandoned label indicates that the RFC has been purposefully set aside.

> While these labels suggest that RFCs move through series of prescribed stages, this isn’t always the case. RFCs are low-process and can be used for all kinds of purposes that may not have an explicit, actionable outcome.

Reviewer: maybe add a sentence here about how the RFC should make it clear (beyond just the label) what the status is and expected process is

> While these labels suggest that RFCs move through series of prescribed stages, this isn’t always the case. RFCs are low-process and can be used for all kinds of purposes that may not have an explicit, actionable outcome.

Author: ```suggestion
While these labels suggest that RFCs move through series of prescribed stages, this isn’t always the case. RFCs are low-process and can be used for all kinds of purposes that may not have an explicit, actionable outcome. Ultimately, the RFC itself should make it clear what the status and planned process is.
```

In this format, we find it helpful to include:

- **Tags** for the people involved and their roles in the decision-making framework (including the [decider](../decisions.md#decider), input providers, approvers, and [approvals](../decisions.md#approvers)).
- **Tags** for any teams that will be affected either directly or indirectly, and should have the opportunity to review and contribute to the RFC.
- **Background**: Just enough context necessary to frame the rest of the RFC. The content should be indisputable facts, not opinion.
- **Problem**: A description of the problem that this RFC is trying to address, the constraints, and why this problem is worth solving now.
- **Proposed solution** (optional): A description of how to solve the problem. It’s okay to skip this section if you just want to define a problem. Otherwise, it can be helpful for thinking of solutions or to hand off ownership to someone else.
- **Definition of success**: How do we know if this proposal was successful? Are there any metrics we need to start tracking?

[Get the template](https://docs.google.com/document/d/1FJ6AhHmVInSE22EHcDZnzvvAd9KfwOkKvFpx7e346z4/edit?usp=sharing) →

> ### Goal: Surfacing a Tension

Reviewer: nit: [style guide](https://about.sourcegraph.com/handbook/communication/content_guidelines/style_and_mechanics#capitalization) indicates we shouldn't capitalize this

Author: ```suggestion
### Goal: Surfacing a tension
```

> This format is helpful for surfacing a Tension. A Tension is somewhere between a problem, a feeling (of unease), and a person’s (unmet) needs. When Tensions come up, they describe a sense there’s a gap between the current reality and an ideal potential future.

Reviewer: It feels weird/inconsistent to capitalize "Tension" here. For example, the previous section isn't "This format is helpful for framing a Problem."

```suggestion
This format is helpful for surfacing a tension. A tension is somewhere between a problem, a feeling (of unease), and a person’s (unmet) needs. When tensions come up, they describe a sense there’s a gap between the current reality and an ideal potential future.
```
